src/Aspire.Hosting/ApplicationModel/IResourceWithConnectionString.cs

```diff
@@ -22,7 +23,7 @@ public interface IResourceWithConnectionString : IResource, IManifestExpressionP
     /// <summary>
     /// Describes the connection string format string used for this resource in the manifest.
     /// </summary>
-    public string? ConnectionStringExpression => null;
+    public ReferenceExpression ConnectionStringExpression { get; }
```

Comment:
It's kind of confusing to have both:

`public ReferenceExpression ConnectionStringExpression`
and
`public string ConnectionStringReferenceExpression`

Can the latter be renamed so these don't sound like the same thing?